lib/email/receiver.rb

```diff
@@ -1359,11 +1359,12 @@ def create_post_with_attachments(options = {})
     def add_attachments(raw, user, options = {})
       raw = raw.dup
 
-      previous_upload_ids =
-        UploadReference
-          .where(target_id: Post.where(topic_id: options[:topic_id]).select(:id))
-          .pluck(:upload_id)
-          .uniq
+      upload_ids =
+        UploadReference.where(
+          target_id: Post.where(topic_id: options[:topic_id]).select(:id),
+        ).pluck("DISTINCT upload_id")
+
+      upload_shas = Upload.where(id: upload_ids).pluck("DISTINCT COALESCE(original_sha1, sha1)")
```

Comment:
Retrieving the `original_sha`s when secure uploads is enabled, otherwise retrieve the `sha`s of the uploads.